src/Orchard.Web/Modules/Orchard.Setup/Services/SetupService.cs

```diff
@@ -154,6 +155,10 @@ public class SetupService : ISetupService {
                     schemaBuilder.AlterTable("Orchard_Framework_DataMigrationRecord",
                         table => table.AddUniqueConstraint("UC_DMR_DataMigrationClass_Version", "DataMigrationClass", "Version"));
 
+                    // Create the distributed lock record schema.
+                    var distributedLockSchemaBuilder = new DistributedLockSchemaBuilder(_shellSettings, schemaBuilder);

```

Comment:
I see what you're saying now. Since we're note using the distributed lock service during setup anyway, we don;t need the lock record table to exist in the first place. After setup has finished, automatic migrations kick in and will take care of creating the required table. No invocation needed from Setup, no need for the `Migrations` class. Will apply the suggestion and test.